pubsub/tests/unit/pubsub_v1/publisher/batch/test_thread.py

```diff
@@ -84,19 +84,32 @@ def test_commit():
     with mock.patch.object(threading, 'Thread', autospec=True) as Thread:
         batch.commit()
 
-        # A thread should have been created to do the actual commit.
-        Thread.assert_called_once_with(
-            name='Thread-CommitBatchPublisher',
-            target=batch._commit,
-        )
-        Thread.return_value.start.assert_called_once_with()
+    # A thread should have been created to do the actual commit.
+    Thread.assert_called_once_with(
+        name='Thread-CommitBatchPublisher',
+        target=batch._commit,
+    )
+    Thread.return_value.start.assert_called_once_with()
```

Comment:
My rationale for this being within the `with` block is that it makes more intuitive and logical sense to me to only use variables declared by context managers within the context manager.

Reply:
I'll put it back.

My rationale: do **as little as possible** while the context manager is active